File: bookstore3/src/service/orderService.js
import config from '../utils/config';
import {postRequest} from '../utils/ajax';

export const getOrders = (data,callback) => {
    const url = `${config.backendUrl}/GetOrders`;
    postRequest(url,data,callback);
}

export const addToOrder = (data, callback) =>{
    const url = `${config.backendUrl}/AddOneOrder`;
    postRequest(url,data,callback);
}

export const getAllOrders = (data, callback) =>{
    const url =`${config.backendUrl}/GetAllOrders`;
    postRequest(url,data,callback);
}

export const getBookSales = (data, callback) =>{
    const url =`${config.backendUrl}/GetBookSales`;
    postRequest(url,data,callback);
}

export const getConsumption = (data, callback) =>{
    const url = `${config.backendUrl}/GetUserConsumption`;
    postRequest(url, data, callback);
}

export const FilterBookByTime = (data, callback) =>{
    const url = `${config.backendUrl}/FilterBookByTime`;
    postRequest(url, data, callback);
}

export const AddToOrders = (data, callback) => {
    const url = `${config.backendUrl}/addToOrder`;
    postRequest(url,data,callback);
}

export const GetOrderDetails = (data, callback) => {
    const url = `${config.backendUrl}/GetOrderDetails`;
    postRequest(url,data,callback);
}

export const FilterTime = (data, callback) => {
    const url = `${config.backendUrl}/FilterTime`;
    postRequest(url,data,callback);
}

export const FilterAllOrderByTime = (data, callback) =>{
    const url = `${config.backendUrl}/FilterAllOrderByTime`;
    postRequest(url,data,callback);
}

export const FilterUserOrderByTime = (data, callback) =>{
    const url = `${config.backendUrl}/FilterUserOrderByTime`;
    postRequest(url,data,callback);
}

export const GetBookSalesById  = (data, callback) =>{
    const url = `${config.backendUrl}/GetBookSalesById`;
    postRequest(url,data,callback);
}

export const FilterUserBookByTime = (data,callback) =>{
    const url = `${config.backendUrl}/FilterUserBookByTime`;
    postRequest(url,data,callback);
}

File: demo1/src/main/java/com/example/demo/service/OrderService.java
package com.example.demo.service;

import com.alibaba.fastjson.JSON;
import com.alibaba.fastjson.JSONObject;
import com.example.demo.entity.OrderItems;
import com.example.demo.entity.Orders;
import org.springframework.web.bind.annotation.RequestBody;

import java.util.List;


public interface OrderService {

    JSONObject AddOneOrder(JSONObject info);

    List<Orders> GetOrders(JSONObject info);

    List<Orders> GetAllOrders();

    List<JSONObject> GetOrderDetails(JSONObject info);

    List<JSONObject> GetBookSales();

    List<JSONObject> GetUserConsumption();

    List<JSONObject> FilterTime(JSONObject info);

    List<JSONObject> FilterBookByTime(JSONObject info);

    JSONObject addToOrder(JSONObject info);

    List<JSONObject> FilterAllOrderByTime(JSONObject info);

    List<JSONObject> FilterUserOrderByTime(JSONObject info);

    JSONObject GetBookSalesById(JSONObject info);

    JSONObject FilterUserBookByTime(JSONObject info);

    void createOneOrder(Orders order, OrderItems orderItems);

    void createOrders(Orders order, List<OrderItems> oi);
}
